lib/honeybadger/plugins/rails.rb

```diff
@@ -29,6 +29,17 @@ def render_exception(arg, exception)
         end
       end
 
+      class ErrorSubscriber
+        def self.report(exception, handled:, severity:, context: {}, source: nil)
+          return if source && ::Honeybadger.config[:'rails.subscriber_ignore_sources'].any? { |regex| regex.match?(source) }
+
+          tags = ['reporter:rails.error_subscriber', "severity:#{severity}", "handled:#{handled}"]
+          tags << "source:#{source}" if source
+          Honeybadger.notify(exception, context: context, tags: tags)
+          exception.instance_variable_set(:@__hb_handled, true)
```

Comment:
This seems like a solid approach at first glance. Rails is now reporting errors from their own instrumentation in some cases, right? So setting this ivar allows our own instrumentation to play nice with theirs?

Reply:
Yep. Without this, some errors would be reported twice.

```ruby
Rails.error.record do
 1 + '1'
end
```

would call our ErrorSubscriber to report the error, then raise the error, which would then be caught by our global error handler middleware and re-reported.